src/App.js

```diff
@@ -177,40 +82,65 @@ class App extends Component {
         </Navbar>
         <Row>
           <Col sm={9}>
-            <SearchForm
-              handleSearchBoxChange={this.handleSearchBoxChange}
-              handleDismissSearchBoxVersion={this.handleDismissSearchBoxVersion}
-              onSubmit={this.handleSubmit}
-              value={this.state.versionInput}
-            />
-            <CurrentRelease
-              version={this.state.version}
-              statuses={this.state.statuses}
-            />
+            <VersionInput />
+            <CurrentRelease />
           </Col>
           <Col sm={3} className="firefox-releases-menu">
             <Panel header={<strong>Firefox Releases</strong>}>
-              <ReleasesMenu versions={this.state.latestChannelVersions} />
+              <SideBar />
             </Panel>
           </Col>
         </Row>
       </Grid>
     );
   }
 }
+const App = connect()(ConnectedApp);
+ConnectedApp.propTypes = {
+  dispatch: PropTypes.func.isRequired,
+};
 
-class SearchForm extends Component {
-  render() {
-    return (
-      <form className="search-form well" onSubmit={this.props.onSubmit}>
-        <ClearableTextInput
-          onChange={this.props.handleSearchBoxChange}
-          onClick={this.props.handleDismissSearchBoxVersion}
-          value={this.props.value}
-        />
-      </form>
-    );
-  }
+const VersionInput = connect(
+  // mapStateToProps
+  state => {
+    return {
+      value: state.versionInput,
+    };
+  },
+  // mapDispatchToProps
+  dispatch => {
```

Comment:
Interesting, I've never seen a `mapDispatchToProps` written out "by hand" before. In kinto-admin they're generated using `bindActionCreators`.

Reply:
ah? Never heard about that, i've just followed the standard way to do thing (I think?) explained in the redux tutorial.